src/model_signing/api.py

```diff
@@ -0,0 +1,582 @@
+
+        Args:
+            hashing_algorithm: the hashing algorithm to use.
+
+        Returns:
+            An instance of the requested hasher.
+        """
+        match hashing_algorithm:
+            case "sha256":
+                return memory.SHA256()
+            case "blake2":
+                return memory.BLAKE2()
+            case _:
+                raise ValueError(
+                    f"Unsupported hashing method {hashing_algorithm}"
+                )
+
+    def _build_file_hasher_factory(
+        self,
+        hashing_algorithm: Literal["sha256", "blake2"] = "sha256",
+        chunk_size: int = 8192,
+    ) -> Callable[[pathlib.Path], file.SimpleFileHasher]:
+        """Builds the hasher factory for a serialization by file.
+
+        Args:
+            hashing_algorithm: the hashing algorithm to use to hash a file
+            chunk_size: The amount of file to read at once. Default is 8KB. A
+              special value of 0 signals to attempt to read everything in a
+              single call.
+
+        Returns:
+            The hasher factory that should be used by the active serialization
+            method.
+        """
+        algorithm = self._build_stream_hasher(hashing_algorithm)
+
+        def factory(path: pathlib.Path) -> file.SimpleFileHasher:
+            return file.SimpleFileHasher(path, algorithm, chunk_size=chunk_size)
+
+        return factory
+
+    def _build_sharded_file_hasher_factory(
+        self,
+        hashing_algorithm: Literal["sha256", "blake2"] = "sha256",
+        chunk_size: int = 8192,
+        shard_size: int = 1000000,
+    ) -> Callable[[pathlib.Path, int, int], file.ShardedFileHasher]:
+        """Builds the hasher factory for a serialization by file shards.
+
+        Args:
+            hashing_algorithm: the hashing algorithm to use to hash a file
+            chunk_size: The amount of file to read at once. Default is 8KB. A
+              special value of 0 signals to attempt to read everything in a
+              single call.
+            shard_size: The size of a file shard. Default is 1,000,000 bytes.
+
+        Returns:
+            The hasher factory that should be used by the active serialization
+            method.
+        """
+        algorithm = self._build_stream_hasher(hashing_algorithm)
+
+        def factory(
+            path: pathlib.Path, start: int, end: int
+        ) -> file.ShardedFileHasher:
+            return file.ShardedFileHasher(
+                path,
+                algorithm,
+                start=start,
+                end=end,
+                chunk_size=chunk_size,
+                shard_size=shard_size,
+            )
+
+        return factory
+
+    def set_serialize_by_file_to_manifest(
+        self,
+        *,
+        hashing_algorithm: Literal["sha256", "blake2"] = "sha256",
+        chunk_size: int = 8192,
+        max_workers: int | None = None,
+        allow_symlinks: bool = False,
+    ) -> Self:
```

Comment:
the default in the `HashingConfig` is to allow symlinks, which conflicts with this. 

I think it would be good to align the defaults one way or another

Reply:
Nice catch. I changed it so even in default configuration symlinks are errors.

Reply:
Thanks. As long as the defaults are consistent :)